autonat.go

```diff
@@ -302,6 +333,30 @@ func (as *AmbientAutoNAT) recordObservation(observation autoNATResult) {
 	}
 }
 
+func (as *AmbientAutoNAT) tryProbe(p peer.ID) bool {
+	as.lastProbeTry = time.Now()
+	if p.Validate() != nil {
+		return false
+	}
+
+	if lastTime, ok := as.recentProbes[p]; ok {
+		if time.Since(lastTime) < as.throttlePeerPeriod {
+			return false
+		}
+	}
+	as.cleanupRecentProbes()
+
+	info := as.host.Peerstore().PeerInfo(p)
+
+	if !as.config.dialPolicy.skipPeer(info.Addrs) {
+		as.recentProbes[p] = time.Now()
+		as.lastProbe = time.Now()
+		go as.probe(&info)
+		return true
+	}
+	return false
+}
+
 func (as *AmbientAutoNAT) probe(pi *peer.AddrInfo) {
```

Comment:
ot: why do we pass an addrinfo here? We should just pass the peer ID.

Reply:
eventually it would be nice to use the address that we've found as the address of the peer that we think is a valid public address we want them to dial us on, and send the autonat request for dialing to that specific multi-addr. E.g. if i'm connected to the peer over a different address (a lan / vpn multiaddr), i'd really like to have the dial/dial-back over the global addresses. That requires more modification of the connection / swarm interface than is likely to happen anytime soon though.

Reply:
Ah, got it.